src/views/Login/Login.tsx

```diff
@@ -21,101 +23,121 @@ const autocompleteSx = {
   width: "100%",
 };
 
-export interface IInputOptions extends InputProps {
+export interface IAddressInputProps extends InputProps {
   localStorageAccounts: Array<string>;
-  inputOnChangeFn: (value: string | null) => void;
+  onInputChange: (value: string | null) => void;
+  placeholderText: string;
 }
 
-const AddressInput: React.FC<IInputOptions> = ({ localStorageAccounts, value, inputOnChangeFn }) => (
-  /*
-   * TODO: Add autosuggest-highlight? It's a small custom package which requires some additional renderOptions
-   * https://mui.com/material-ui/react-autocomplete/#Highlights.tsx
-   */
-
+const AddressInput: React.FC<IAddressInputProps> = ({ localStorageAccounts, value, onInputChange, placeholderText }) => (
   <StyledAutocomplete
     sx={autocompleteSx}
     value={value}
     freeSolo
     disablePortal
-    onChange={(e, value) => inputOnChangeFn(value as string)}
+    onChange={(e, value) => onInputChange(value as string)}
     options={localStorageAccounts}
-    renderInput={(params) => <TextField onChange={(e) => inputOnChangeFn(e.target.value)} {...params} label="Enter Account" />}
+    renderInput={(params) => <TextField onChange={(e) => onInputChange(e.target.value)} {...params} label={placeholderText} />}
   />
 );
 
 const Login: React.FC = () => {
   const { flushFn, userLogin } = useAccount();
   const [accounts, setAccounts] = useLocalStorage<Array<string>>("accounts", []); // Stores user accounts in localStorage under "accounts" key
   const [existingUser, setExistingUser] = useState<"existing" | "new">("new");
+  const [existingUserType, setExistingUserType] = useState<"address" | "secretPhrase">("address");
   const [userInputAccount, setUserInputAccount] = useState<string>("");
-  const [isValidAddressState, setIsValidAddressState] = useState<boolean>(false);
+  const [isValidInputState, setIsValidInputState] = useState<boolean>(false);
   const [userRememberState, setUserRememberState] = useState<boolean>(false);
   const { enqueueSnackbar } = useSnackbar();
 
-  const handleAddressInput = useCallback((newValue: string | null) => {
-    if (newValue === null) {
-      return;
-    }
-
-    if (!isValidAddress(newValue)) {
-      setIsValidAddressState(false);
-    } else {
-      setIsValidAddressState(true);
-    }
-
-    setUserInputAccount(newValue);
-
-    /*
-     * TODO: Add formatting to address entry
-     * - Auto uppercase
-     * - 'JUP-' prefixing and hyphens at appropriate positions
-     *
-     * if (newValue.length > 2) {
-     *   newValue += "-";
-     * }
-     */
-  }, []);
+  const handleAddressOrSecretInput = useCallback(
+    (newValue: string | null) => {
+      if (newValue === null) {
+        return;
+      }
+      setUserInputAccount(newValue);
+
+      if (existingUserType === "address") {
```

Comment:
same comment as my other statement about the `handleLogin` callback. you wouldn't have to refactor this either, it's a suggestion for a bit easier to digest code.

Reply:
To me this one is a bit more readable than the other one I refactored. This Login.tsx component is already 260 lines long so making it even longer to refactor this function as well may not be a good tradeoff?

Reply:
this one is much much shorter already than the other one. so i don't see a problem leaving it as-is. it also works as-is so no need to fix things that aren't broken.